one_fm/overrides/leave_allocation.py

```diff
@@ -0,0 +1,27 @@
+import frappe
+
+from hrms.hr.doctype.leave_allocation.leave_allocation import (
+    LeaveAllocation
+)
+from frappe import _,  bold
+
+
+class LeaveAllocationOverride(LeaveAllocation):
+	# overide default class
+
+	def validate(self):
+		super().validate()
+		self.validate_maternity_leave()
+		
+	def validate_maternity_leave(self):
+		leave_type = frappe.get_doc("Leave Type", self.leave_type)
+		employee = frappe.get_doc("Employee", self.employee)
```

Comment:
As regards this, since you are tryng to get the gneder, you could just use the, db.get_value method, instead of taking all columns

Reply:
Alright